modules/database/export.bal

```diff
@@ -272,40 +279,89 @@ public isolated transactional function getTimelineStepLimit(int experimentId) re
 
 }
 
-public isolated transactional function getExperimentDBExport(int experimentId) returns ExperimentCompleteExport|error {
+public isolated transactional function getExperimentDBExport(int experimentId, ExperimentExportConfig config) returns ExperimentCompleteExport|error {
     TimelineStepExport[] timelineSteps = [];
     ExperimentDataExport[] experimentDataList = [];
     ExperimentFull experiment = check getExperiment(experimentId);
     int[] dataIdList = [];
 
     // iterate over timeline steps
-
     TimelineStepFull[] timelineStepListDb = check getTimelineStepList(experimentId, (), (), (), (), allAttributes = true, 'limit = check getTimelineStepLimit(experimentId));
+
+    int[] stepList = [];
+    if config.restriction == "STEPS" {
+        // TODO: check that timelineStepListDb is sorted
+        stepList = config.stepList.sort();
+    }
+    int counter = 0;
+    boolean add;
     foreach TimelineStepFull timelineStepDb in timelineStepListDb {
-        TimelineStepExport timelineStepExport = check castToTimelineStepExport(timelineStepDb);
-        // retrieve associated step data
-        timelineStepExport.stepDataList = check getTimelineStepDataList(timelineStepDb.stepId);
-        foreach var stepData in timelineStepExport.stepDataList {
-            dataIdList.push(stepData.dataId);
+        add = true;
+        if config.restriction == "STEPS" && timelineStepDb.sequence == stepList[counter] {
+            // filter steps with stepList 
+            if timelineStepDb.sequence == stepList[counter] {
+                counter += 1;
+            } else {
+                add = false;
+            }
         }
-        // retrieve associated substeps with their substep data
-        timelineStepExport.timelineSubsteps = check getTimelineSubstepsExport(timelineStepDb.stepId);
-        timelineSteps.push(timelineStepExport);
-        foreach var substep in timelineStepExport.timelineSubsteps {
-            foreach var substepData in substep.substepDataList {
-                dataIdList.push(substepData.dataId);
+        if add {
+            TimelineStepExport timelineStepExport = check castToTimelineStepExport(timelineStepDb);
+            if config.restriction != "LOGS" {
+                // retrieve associated step data
+                timelineStepExport.stepDataList = check getTimelineStepDataList(timelineStepDb.stepId);
+                foreach var stepData in timelineStepExport.stepDataList {
+                    dataIdList.push(stepData.dataId);
+                }
+            } else {
+                // don't need data for restriction "LOGS"
+                timelineStepExport.stepDataList = [];
```

Comment:
I try to keep conditions as readable as possible (which is subjective). In this case I suggest switching the cases. The easier case first is (in my opinion) easier to parse and then forget. It also changes the negative != test into a positive == test (negations are sometimes harder to understand).